lib/spack/spack/cmd/logs.py

```diff
@@ -0,0 +1,85 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+import os.path
+
+import llnl.util.tty as tty
+import spack
+import spack.cmd
+import spack.cmd.common.arguments as arguments
+
+
+description = "fetch the logs of an installed spec"
+section = "basic"
+level = "short"
+
+
+def setup_parser(subparser):
+    subparser.add_argument(
+        '--environment',
```

Comment:
I would call this `--build-env` or `--build-environment`.  I'm somewhat conflicted about whether this should be an argument to this command or to `spack env`, e.g. `spack env --installed <spec>`.  The latter would also let you actually run a command in the old environment, which could be very useful.

Reply:
Let me know if you want this command at all, or if you want just want 
```
spack env --install <spec>
```

For the what is worth, the motivation that lead me putting this PR together was just being tired to repeat:
```
$ spack find -p <spec>
==> 1 installed packages.
-- linux-ubuntu14.04-x86_64 / gcc@4.8 ---------------------------
    zlib@1.2.11  /home/mculpo/PycharmProjects/spack/opt/spack/linux-ubuntu14.04-x86_64/gcc-4.8/zlib-1.2.11-eksallf6cymqkp6pkz6ymzjakqt6bqkx

$ less <cut'n paste of the path above>/.spack/build.out
```
over and over when looking at previous installations logs. Having a dump of the environment was just a low hanging fruit once opening the logs had been implemented.

Reply:
How about just these:

* `spack env --install <spec>`
* `spack edit --log <spec>`

or just the latter if you're short on time?  I think the use case you describe is a good one, we just already have `spack edit`

Reply:
PS -- `spack edit` probably only need another arg and a call to `disambiguate spec` in `edit_package` to make this work